File: typecast-java/src/main/java/io/typecast/models/GenderEnum.java
package io.typecast.models;

/**
 * Gender enumeration for voice filtering.
 */
public enum GenderEnum {
    /** Male voice */
    MALE("male"),
    /** Female voice */
    FEMALE("female");

    private final String value;

    GenderEnum(String value) {
        this.value = value;
    }

    /**
     * Returns the string value of the gender.
     * 
     * @return the gender string
     */
    public String getValue() {
        return value;
    }

    /**
     * Creates a GenderEnum from a string value.
     * 
     * @param value the gender string
     * @return the corresponding GenderEnum
     * @throws IllegalArgumentException if the value doesn't match any gender
     */
    public static GenderEnum fromValue(String value) {
        if (value == null) {
            return null;
        }
        for (GenderEnum gender : values()) {
            if (gender.value.equalsIgnoreCase(value)) {
                return gender;
            }
        }
        throw new IllegalArgumentException("Unknown gender: " + value);
    }

    @Override
    public String toString() {
        return value;
    }
}

File: typecast-csharp/src/Typecast/Models/GenderEnum.cs
using System.Text.Json.Serialization;

namespace Typecast.Models;

/// <summary>
/// Gender classification for voice filtering.
/// </summary>
[JsonConverter(typeof(JsonStringEnumMemberConverter))]
public enum GenderEnum
{
    /// <summary>Male voice</summary>
    [JsonPropertyName("male")] Male,
    
    /// <summary>Female voice</summary>
    [JsonPropertyName("female")] Female
}

/// <summary>
/// Extension methods for GenderEnum.
/// </summary>
public static class GenderEnumExtensions
{
    /// <summary>
    /// Converts the GenderEnum to its API string representation.
    /// </summary>
    public static string ToApiString(this GenderEnum gender) => gender switch
    {
        GenderEnum.Male => "male",
        GenderEnum.Female => "female",
        _ => throw new ArgumentOutOfRangeException(nameof(gender))
    };

    /// <summary>
    /// Parses a string to GenderEnum.
    /// </summary>
    public static GenderEnum ParseGender(string value) => value.ToLowerInvariant() switch
    {
        "male" => GenderEnum.Male,
        "female" => GenderEnum.Female,
        _ => throw new ArgumentException($"Unknown gender: {value}", nameof(value))
    };
}
